README.md

```diff
@@ -99,20 +86,56 @@ Note that you may not get cached results if you use a different `nix` version th
 
 ## Building arbitrary packages
 
-The [`survey`](./survey) directory maintains a select set of Haskell executables that are known and not known to work with this approach; contributions are welcome to grow the set of working executables.
-Run for example:
+The [`survey/default.nix`](./survey/default.nix) was originally a survey of Haskell executables that are known to (and known _not_ to) work with this approach, however it also exposes packages sets with overridden Haskell packages and dependencies that you can use to build _your own_ packages. The name `survey` shouldn't put you off from using it.
+
+If you are a nix user, you can `import` this functionality and override the `haskellPackages` to include your own package, for example in the [PostgREST project](https://github.com/PostgREST/postgrest/blob/main/nix/static-haskell-package.nix).
+
+### Structure of `survey/default.nix`
+
+The process of building up to the final `haskellPackages` is broken down into multiple steps, with the intermediate package sets also exposed for your use.
+
+We start with a plain nixpkgs named `normalPkgs`, defaulting to the version provided by [`nixpkgs.nix`](./nixpkgs.nix), but you can pass in your own version here.
+You could instead provide `overlays`, which get applied to `normalPkgs`.
+The `.pkgsMusl` from `normalPkgs` now forms our base `pkgs`.
+
```

Comment:
I propose to format the below as a list of the overlays and exposed package sets, and then for each explain its goal as you do, e.g. with this structure:

```
* `archiveFilesOverlay`
  * exposed package set: `archiveFilesOverlay`
  * **Goal:** Packages in `pkgsMusl` typically [..]
* `haskellLibsReadyForStaticLinkingOverlay`
  * exposed package set: `pkgsWithHaskellLibsReadyForStaticLinking`
* `staticHaskellBinariesOverlay`
  * ...
* `pkgsWithStaticHaskellBinaries`
  * ...
```